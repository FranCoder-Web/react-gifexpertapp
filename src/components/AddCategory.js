import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('')

	const handleInputChange = (e) => {
		setInputValue(e.target.value)
	}
	const handleSubmit = (e) => {
		e.preventDefault()

		if (inputValue.trim().length > 2) {
			setCategories((prevState) => {
				let repeatCategorie = prevState.indexOf(
					inputValue.toUpperCase().trim()
				)
				if (repeatCategorie >= 0) {
					return prevState
				} else {
					return [inputValue.toUpperCase(), ...prevState]
				}
			})
			setInputValue('')
		}
	}
	return (
		<form onSubmit={handleSubmit}>
			<input
				className='search'
				type='text'
				value={inputValue}
				onChange={handleInputChange}
				placeholder='SEARCH GIFS...'
			/>
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
}
