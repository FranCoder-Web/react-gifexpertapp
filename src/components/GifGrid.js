import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import Loader from 'react-loader-spinner'

export const GifGrid = ({ category, cleanCategory }) => {
	const { data: images, loading } = useFetchGifs(category)
	const handleCategory = () => {
		cleanCategory((prevstate) => {
			return prevstate.filter((cat) => cat !== category)
		})
	}

	return (
		<>
			<h2 className='category'>
				{category}
				<span onClick={handleCategory} className='x-category'>
					X
				</span>
			</h2>
			{loading && (
				<Loader
					type='ThreeDots'
					color='#1e272e'
					height={120}
					width={120}
				/>
			)}
			<div className='card-grid'>
				{images.map((img) => (
					<GifGridItem key={img.id} category={category} {...img} />
				))}
			</div>
		</>
	)
}

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
}
