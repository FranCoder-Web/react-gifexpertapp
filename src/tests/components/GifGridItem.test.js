import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Probando el componente <GifGridItem />', () => {
	const title = 'UN TITULO'
	const url = 'http://localhost/algo.jpg'
	const wrapper = shallow(<GifGridItem title={title} url={url} />)

	test('debe mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot()
	})

	test('debe tener el parrafo con el "title"', () => {
		const p = wrapper.find('p')

		expect(p.text()).toBe(title)
	})

	test('debe tener una img con url y alt ', () => {
		const img = wrapper.find('img')

		expect(img.prop('src')).toBe(url)
		expect(img.prop('alt')).toBe(title)
	})

	test('debe contener la clase animate__fadeInDown', () => {
		const className = wrapper.find('div').prop('className')

		expect(className.includes('animate__fadeInDown')).toBe(true)
	})
})
