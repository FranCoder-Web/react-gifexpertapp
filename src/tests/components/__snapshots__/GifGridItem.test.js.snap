// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown "
>
  <img
    alt="UN TITULO"
    src="http://localhost/algo.jpg"
  />
  <p>
    UN TITULO
  </p>
</div>
`;
